egs/ami/s5/local/ami_ihm_scoring_data_prep.sh

```diff
@@ -87,19 +87,12 @@ sort -k 2 $dir/utt2spk | utils/utt2spk_to_spk2utt.pl > $dir/spk2utt || exit 1;
 join $dir/utt2spk $dir/segments | \
    perl -ne '{BEGIN{$pu=""; $pt=0.0;} split;
            if ($pu eq $_[1] && $pt > $_[3]) {
-             print "$_[0] $_[2] $_[3] $_[4]>$_[0] $_[2] $pt $_[4]\n"
+             print "s/^$_[0] $_[2] $_[3] $_[4]\$/$_[0] $_[2] $pt $_[4]/;\n"
            }
-           $pu=$_[1]; $pt=$_[4]; 
+           $pu=$_[1]; $pt=$_[4];
          }' > $dir/segments_to_fix
-if [ `cat $dir/segments_to_fix | wc -l` -gt 0 ]; then
-  echo "$0. Applying following fixes to segments"
-  cat $dir/segments_to_fix
```

Comment:
Maybe it makes better sense to leave the diagnostic output? It's in a different format, but still as human-readable as before. The test can be simplified to `if [ -s $dir/segments_to_fix ]` (`-s` test for for non-empty file).